dashbuilder/dashbuilder-shared/dashbuilder-dsl/src/main/java/org/dashbuilder/dsl/serialization/impl/DashboardZipSerializer.java

```diff
@@ -0,0 +1,259 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+import java.util.zip.ZipOutputStream;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import org.dashbuilder.dataset.def.CSVDataSetDef;
+import org.dashbuilder.dataset.def.DataSetDef;
+import org.dashbuilder.dataset.json.DataSetDefJSONMarshaller;
+import org.dashbuilder.dsl.factory.dashboard.DashboardFactory;
+import org.dashbuilder.dsl.factory.navigation.NavigationFactory;
+import org.dashbuilder.dsl.model.Dashboard;
+import org.dashbuilder.dsl.model.Navigation;
+import org.dashbuilder.dsl.model.Page;
+import org.dashbuilder.dsl.serialization.DashboardSerializer;
+import org.dashbuilder.external.model.ExternalComponent;
+import org.dashbuilder.navigation.json.NavTreeJSONMarshaller;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.uberfire.ext.layout.editor.api.editor.LayoutTemplate;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.dashbuilder.dsl.helper.ComponentsHelper.listComponents;
+import static org.dashbuilder.dsl.helper.ComponentsHelper.listPagesComponents;
+
+public class DashboardZipSerializer implements DashboardSerializer {
+
+    Logger logger = LoggerFactory.getLogger(DashboardZipSerializer.class);
+
+    private static final String BASE_PATH = "dashbuilder/";
+
+    private static final String LAYOUTS_PATH = BASE_PATH + "perspectives/";
+    private static final String DATA_SETS_BASE = BASE_PATH + "datasets/";
+    private static final String COMPONENTS_BASE = BASE_PATH + "components/";
+    private static final String DATA_SETS_PATH = DATA_SETS_BASE + "definitions/";
+    private static final String NAVIGATION_PATH = BASE_PATH + "navigation/navigation/navtree.json";
+
+    private static final DataSetDefJSONMarshaller DATA_SET_MARSHALLER = new DataSetDefJSONMarshaller(new InternalDataSetProviderRegistry());
+
+    private static final Gson gson = new GsonBuilder().create();
+
+    /**
+     * Does not support: CSV files and components files
+     */
+    @Override
+    public Dashboard deserialize(InputStream model) {
+        Map<String, String> importContent = readAllEntriesContent(model);
+
+        List<Page> pages = new ArrayList<>();
+        List<DataSetDef> datasets = new ArrayList<>();
+        AtomicReference<Navigation> navigationRef = new AtomicReference<>(NavigationFactory.emptyNavigation());
+
+        importContent.forEach((path, content) -> {
+            if (path.startsWith(LAYOUTS_PATH) && path.endsWith("perspective_layout")) {
+                LayoutTemplate template = gson.fromJson(content, LayoutTemplate.class);
+                pages.add(Page.create(template));
+            }
+
+            if (path.startsWith(DATA_SETS_PATH) && path.endsWith("dset")) {
```

Comment:
Consider extracting `"dset"` and `"perspective_layout"` into constants.
Lines 190 and 198 would need them.